host_source.go

```diff
@@ -413,8 +414,9 @@ func (h *HostInfo) update(from *HostInfo) {
 	if h.dataCenter == "" {
 		h.dataCenter = from.dataCenter
 	}
-	if h.rack == "" {
+	if h.missingRack || h.rack == "" {
```

Comment:
Maybe we should remove the `h.rack == ""` from here? If now we treat `""` as a valid rack.